sdk/eventhub/mock-hub/src/messages/event-hubs/partitionInfo.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Message, types } from "rhea";
+
+/**
+ * Checks whether the provided message is requesting the partition info from the Event Hub.
+ * @param entityPath The path the client sent the request to.
+ * Expected to be `$management` if the message is requesting runtime info.
+ * @param message The message sent by the client.
+ */
+export function isPartitionInfo(entityPath: string, message: Message): boolean {
+  if (entityPath !== "$management") {
+    return false;
+  }
+
+  const properties = message.application_properties;
+  if (!properties) {
+    return false;
+  }
+
+  if (
+    properties.operation !== "READ" ||
+    properties.type !== "com.microsoft:partition" ||
+    typeof properties.partition !== "string"
+  ) {
+    return false;
+  }
+
+  if (!message.body) {
+    return true;
+  }
+
+  try {
+    const body = JSON.parse(message.body);
+    if (Array.isArray(body) && !body.length) {
+      return true;
+    }
+  } catch (err) {
+    return false;
+  }
+  return false;
+}
+
+export interface GeneratePartitionInfoResponseOptions {
+  correlationId?: string;
+  eventHubName: string;
+  targetLinkName?: string;
+  beginningSequenceNumber: number;
+  lastEnqueuedSequenceNumber: number;
+  lastEnqueuedOffset: string;
+  lastEnqueuedTimeUtc: Date;
+  isPartitionEmpty: boolean;
+  partitionId: string;
+}
+
+/**
+ * Generates a message containing the EventHub's specified partition info.
+ */
+export function generatePartitionInfoResponse({
+  eventHubName,
+  correlationId,
+  targetLinkName,
+  beginningSequenceNumber,
+  lastEnqueuedSequenceNumber,
+  lastEnqueuedOffset,
+  lastEnqueuedTimeUtc,
+  isPartitionEmpty,
+  partitionId
+}: GeneratePartitionInfoResponseOptions): Message {
+  return {
+    to: targetLinkName,
+    correlation_id: correlationId,
+    application_properties: { operation: "READ", "status-code": types.wrap_int(200) },
+    body: {
+      name: eventHubName,
+      type: "com.microsoft:partition",
+      begin_sequence_number: types.wrap_long(beginningSequenceNumber),
```

Comment:
Was this the 'unsigned/signed' issue you were talking about the other day?

Reply:
Sort of. These really are supposed to be `long`. It was the status-code that rhea was treating as an unsigned int, but Java (and I think .NET) were expecting an int.